FLEDGE.md

```diff
@@ -437,12 +441,56 @@ The bundle may be fetched using credentials like cookies, as described in the [s
 
 Worklet processes follow Chrome's standard site-isolation policies. On Android, due to resource constraints, it's possible that a worklet may run in the same process as a renderer.
 
-##### CORS Required
+###### CORS Required
 
 The origin of the frame that called `runAdAuction()` is not required to match the origin of `directFromSellerSignals`, so CORS is used when fetching `directFromSellerSignals`.  This means subresources in the bundle should include the [Access-Control-Allow-Origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin) header to authorize the origin that calls `runAdAuction()` -- this should also be done on the bundle file itself.
 
 For the JSON response, only the `https` scheme is supported -- the `uuid-in-package` scheme isn't supported as that scheme doesn't support CORS.
 
+#### 2.5.2 Using Response Headers
+
+An alternative way to pass DirectFromSellerSignals without subresource bundles is via the `Ad-Auction-Signals` response header of some `fetch()` request, together with the `directFromSellerSignalsHeaderAdSlot` parameter on `navigator.runAdAuction()`.
+
+With this method, a [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) call is made by some script on the page (this call may be made in a subframe), with an extra option, `{adAuctionHeaders: true}`:
+
+```javascript
+let fetchResponse = await fetch("https://seller.com/signals", {adAuctionHeaders: true});
+```
+
+The browser will make the request it would without `{adAuctionHeaders: true}`, with the exception that the request will also include a request header, `Sec-Ad-Auction-Fetch: ?1`. This header indicates to the server that any `Ad-Auction-Signals` response header from the server will only be loaded in auctions via `directFromSellerSignalsHeaderAdSlot` (this is analogous to the guarantees of `X-FLEDGE-Auction-Only` and `Sec-Fetch-Dest: webbundle` from the [subresource bundle version](#251-using-subresource-bundles) -- scripts on the page cannot set the `Sec-Ad-Auction-Fetch: ?1` request header without using the `{adAuctionHeaders: true}` option).
+
+The value of the Ad-Auction-Signals header should be JSON formatted, with the following schema:
+
+```json
+Ad-Auction-Signals=[{
+  "adSlot": "adSlot/1",
+  "sellerSignals": {/*...*/},
```

Comment:
We probably want to define what types are expected for each of these keys, and what to do if they're wrong.

Reply:
Error handling is described in the spec, but I'm not sure it's worth detailing in the explainer (this document)?

I removed the braces, since the sellerSignals, auctionSignals, and perBuyerSignals value can be of any JSON value type.

The adSlot value is a string, as in the example.

Reply:
> Error handling is described in the spec, but I'm not sure it's worth detailing in the explainer (this document)?

Oops, yeah. I kinda forgot which doc I was reading...